File: push/push.go
package push

import "github.com/aura-studio/dynamic-node-cli/config"

// PushForProcedure executes push operations for a given procedure.
func PushForProcedure(proc config.Procedure) {
	tl := NewTaskList(proc)
	for remote, tasks := range tl.Tasks {
		r := NewS3Remote(remote)
		r.Push(tasks)
	}
}

File: src/push/push.js
import { newTaskList } from "./task-list.js";
import { S3Remote } from "./remote.js";

export async function pushForProcedure(proc) {
  const taskList = await newTaskList(proc);
  for (const [remote, tasks] of taskList.tasks) {
    const target = new S3Remote(remote);
    await target.push(tasks);
  }
}
